src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -202,42 +199,98 @@ public int getDefaultCloudIndexPrefetchBufferSize() {
      */
     @Override
     public void onStartup() {
+        assertVariantPathsOrSampleNameFileWasSpecified();
+        initializeHeaderAndSampleMappings();
+        initializeIntervals();
+        super.onStartup();
+    }
 
-        List<VCFHeader> headers = new ArrayList<>(variantPaths.size());
-
-        for (String variantPath : variantPaths) {
-
-            final AbstractFeatureReader<VariantContext, LineIterator> reader = getReaderFromVCFUri(variantPath);
-            VCFHeader header = (VCFHeader)reader.getHeader();
+    private void assertVariantPathsOrSampleNameFileWasSpecified(){
+        if ( (variantPaths == null || variantPaths.isEmpty()) && sampleNameToPathFile == null) {
+            throw new CommandLineException.MissingArgument(StandardArgumentDefinitions.VARIANT_LONG_NAME,
+                                                       "One of " + StandardArgumentDefinitions.VARIANT_LONG_NAME + " or " + SAMPLE_NAME_MAP_LONG_NAME + " must be specified" );
+        }
+    }
 
-            // A GVCF file must contain only one sample, throw an exception otherwise
-            if (header.getNGenotypeSamples() != 1) {
-                throw new UserException("Input GVCF: " + variantPath + " should contain data for one sample");
-            }
-            String sampleName = header.getGenotypeSamples().get(0);
-            headers.add(header);
-            sampleNames.add(sampleName);
-            if (sampleNameToVcfUri.put(sampleName, variantPath) != null) {
-                throw new UserException("Duplicate sample " + sampleName);
+    private void initializeHeaderAndSampleMappings() {
+        // Only one of -V and -sampleNameToPathFile may be specified
+        if (sampleNameToPathFile == null) {
+            // -V was specified
+            final List<VCFHeader> headers = new ArrayList<>(variantPaths.size());
+
+            for (final String variantPath : variantPaths) {
+                final  VCFHeader header = loadHeaderFromVCFUri(variantPath);
+
+                // A GVCF file must contain only one sample, throw an exception otherwise
+                assertGVCFHasOnlyOneSample(variantPath, header);
+                headers.add(header);
+
+                final String sampleName = header.getGenotypeSamples().get(0);
+                sampleNames.add(sampleName);
+                final String previousPath = sampleNameToVcfUri.put(sampleName, variantPath);
+                if (previousPath != null) {
+                    throw new UserException("Duplicate sample: " + sampleName + ". Sample was found in both "
+                                                    + variantPath + " and " + previousPath + ".");
+                }
             }
+            mergedHeaderLines = VCFUtils.smartMergeHeaders(headers, true);
+            mergedHeaderSequenceDictionary = new VCFHeader(mergedHeaderLines).getSequenceDictionary();
```

Comment:
Should we throw if the merged header has no sequence dictionary (theoretically possible)